### PR TITLE
2 implement adc interface

### PR DESCRIPTION
**Pull request by stevegtaylor**
ADC is functional on continuous mode. Python file added for quick and dirty calibration.

### CODE REVIEW

**Comment by jmadden173** [comment]
Fix the conflicts and ping me to review the code

**Comment by jmadden173** [comment]
@stevegtaylor any update on the conflicts?

**Comment by stevegtaylor** [comment]
I think I need your approval @jmadden173 to finish merging.

I am getting some build errors that I think will be easier to fix once the merge is complete and I can diagnose the problems in vscode.

**Comment by jmadden173** [comment]
I'll see if I can get around to test the version today. @stevegtaylor was build errors are you getting? Can you paste the platformio logs here so I know what to expect when I test?

**Comment by jmadden173** [comment]
Waiting for implementation of reading both voltage and current channels to approve PR

File: stm32/stm32.ioc
Comment (jmadden173) [inline stm32/stm32.ioc:1]: Were going to need to figure out how to effectively resolve IOC file conflicts in the future. In normal dev we wouldn't have this many changes and it would be rare that multiple people change the same IO after its already been configured. Either way this has been a huge headache throughout the dev.

**Review by jmadden173 — APPROVED** [review]
Looks fine. Mainly concerned about the code being copied for ADS reads/write and namespacing of the function calls. I'm approving the PR since I'm expected a code refactor in the future after we complete #32

**Pull request merged**
